src/vm/methodtable.cpp

```diff
@@ -2355,6 +2358,8 @@ bool MethodTable::ClassifyEightBytes(SystemVStructRegisterPassingHelperPtr helpe
     }
     CONTRACTL_END;
     
+    _ASSERTE(!isNativeStruct);
```

Comment:
I don't think that this bool parameter is useful, as the caller could easily be buggy and pass an inaccurate value. Can we not just use something like `_ASSERTE(!HasLayout());` (or a similar condition if that's not precise enough)?